### ISSUE TITLE
`TOKEN_DURATION=0` as environment variable does not create a token without expiration

### ISSUE THREAD

**Issue by evan-barentin** [Kitware/CDash#1269]
Hi,

I’m running a private instance of cdash using the `kitware/cdash:v3.1.0` docker image. In the command line starting the container, the argument `-e TOKEN_DURATION=0` is given to create tokens without expiration.

But instead when a token was created, the expiration date was actually the creation date & hour. After some investigation, I think the problem might be in `AuthToken.php:62` were the comparison between TOKEN_DURATION and 0 is done with `===` :

```php
    public function Generate()
    {
        $duratio n= config('cdash.token_duration');
        $now = time();
        $token = bin2hex(random_bytes(16));
        $this->Hash = $this->HashToken($token);
        $this->Created = gmdate(FMT_DATETIME, $now);
        if ($duration === 0) {
            // Special value meaning "this token never expires".
            $this->Expires = '2038-01-18 03:14:07';
        } else {
            $this->Expires = gmdate(FMT_DATETIME, $now + $duration);
        }
        return $token;
    }
```

As `$duration` is recovered from an environment variable, it’s value is `"0"` and not an integer,  so the `===` does not work as expected.

I hope this could help you, and if you have any question do not hesitate to ask.

**Comment by williamjallen** [comment]
Closed by #1296

**Comment by evan-barentin** [comment]
Thanks for the fix !

Have a nice day :)
